stdlib/Dates/test/periods.jl

```diff
@@ -467,4 +467,21 @@ end
     @test Dates.toms(Dates.Second(1) + Dates.Microsecond(1)) == 1e3
 end
 
+@testset "CompoundPeriod and Period isless()" begin
+    #tests for allowed comparisons
+    #FixedPeriod
+    @test (h - ms < h + ns) == true
+    @test (h + ns < h -ms) == false
+    @test (h  < h -ms) == false
+    @test (h-ms  < h) == true
+    #OtherPeriod
+    @test (2y-m < 25m+1y) == true
+    @test (2y < 25m+1y) == true
+    @test (25m+1y < 2y) == false
+    #tests for error throwing for not allowed comparisons (FixedPeriod and OtherPeriod combinations)
+    @test_throws ErrorException y + h < h + m
```

Comment:
Yeah i agree it is true, i just didnt want to create a comparision that is different to how they seem to be treating other comparisions between the types. 
Id rather it allowed the comparision as id hate getting a random error when it is obvious that Year(1) + Hour(1) > Month(1) but it would be weird if it was allowed but then if you tried to test if Year(1) < Hour(1) it gave an error.
I feel making it that the error being thrown is the normal behaviour and just put somewhere that to get around the error do:
tons(Year(1) + Hour(1)) < tons(Month(1))

Reply:
Sorry to bring up this older pull, but which do you think is the better idea? That error throwing is just a bit of an issue and its hard to tell whether this should follow throwing errors or simply going with what is most likely  and use "tons()"

Reply:
I think we should use `tons`

Reply:
I've just updated it to use tons and the tests